autopts/ptscontrol.py

```diff
@@ -289,10 +289,16 @@ def wait_until():
         finally:
             self._response.clear()
 
-        # Stringify response
-        rsp = str(rsp)
-        rsp_len = str(len(rsp))
-        is_present = str(1)
+        if rsp in ['No', 'Cancel', 'Abort', None]:
+            # NULL pointer is acceptable for any negative response which includes No and Cancel.
+            rsp = ""
+            rsp_len = 0
+            is_present = False
+        else:  # if rsp in ['OK', 'Yes', 'Retry' or Edit value]:
+            # Stringify response
+            rsp = str(rsp)
+            rsp_len = str(len(rsp))
+            is_present = str(1)
```

Comment:
Since you have `rsp = str(rsp)` it indicates that `rsp` is not a string already, and in that case how is `if rsp in ['No', 'Cancel', 'Abort', None]:` expected to work?

Reply:
`rsp = str(rsp)` turns everything into a string

Reply:
> `rsp = str(rsp)` turns everything into a string

Either `rsp` is a string and you can use `if rsp in ['No', 'Cancel', 'Abort', None]:`, or `rsp` is not a string and you cannot use `if rsp in ['No', 'Cancel', 'Abort', None]:`, as it won't be able to compare e.g. `rsp` with `'No'`. 

So as I see it, you have to do `rsp = str(rsp)` before `if rsp in ['No', 'Cancel', 'Abort', None]:` - Or if that is not needed, then `rsp = str(rsp)` shouldn't be needed either.

Reply:
```python
rsp = None

rsp = str(None)
if rsp in ['No', 'Cancel', 'Abort', None]:
    print("OK")
else:
    print("NOT OK")
```

Reply:
I think I need a bit more context - Can you clarify? 

With the above it should probably also be `if rsp in ['No', 'Cancel', 'Abort', 'None']:`, rigth? As `str(None)` returns `'None'`